app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.java

```diff
@@ -0,0 +1,196 @@
+package de.westnordost.streetcomplete.quests.max_height;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.quests.AbstractQuestFormAnswerFragment;
+import de.westnordost.streetcomplete.view.dialogs.AlertDialogBuilder;
+
+public class AddMaxHeightForm extends AbstractQuestFormAnswerFragment
+{
+	public static final String
+		MAX_HEIGHT = "max_height",
+		NO_SIGN = "no_sign";
+
+	private EditText heightInput, feetInput, inchInput;
+
+	private boolean isMetric;
+
+	@Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
+									   Bundle savedInstanceState)
+	{
+		View view = super.onCreateView(inflater, container, savedInstanceState);
+
+		isMetric = getCountryInfo().getHeightUnits().get(0).equals("m");
+
+		setMaxHeightSignLayout(getMaxHeightLayoutResourceId());
+		addOtherAnswers();
+
+		return view;
+	}
+
+	private void setMaxHeightSignLayout(int resourceId)
+	{
+		View contentView = setContentView(resourceId);
+
+		heightInput = contentView.findViewById(R.id.maxHeightInput);
+		feetInput = contentView.findViewById(R.id.maxHeightFeetInput);
+		inchInput = contentView.findViewById(R.id.maxHeightInchInput);
+	}
+
+	private int getMaxHeightLayoutResourceId()
+	{
+		return isMetric ? R.layout.quest_max_height : R.layout.quest_max_height_us;
+	}
+
+	private void addOtherAnswers()
+	{
+
+		if (getCountryInfo().getHeightUnits().size() == 2)
+		{
+			if (isMetric)
+			{
+				addOtherAnswer(R.string.quest_maxheight_answer_imperial_unit, () ->
+				{
+					isMetric = false;
+					setMaxHeightSignLayout(getMaxHeightLayoutResourceId());
+				});
+			}
+			else
+			{
+				addOtherAnswer(R.string.quest_maxheight_answer_metric_unit, () ->
+				{
+					isMetric = true;
+					setMaxHeightSignLayout(getMaxHeightLayoutResourceId());
+				});
+			}
```

Comment:
Okay, that is also a possible solution. Why though did you not solve it in the same way as in the maxspeed quest though? To me, it looks like the swifter solution.

Reply:
I had some trouble with displaying the spinner on my API 19 emulator which I could not yet resolve... For the max speed quest everything worked fine but for this quest it always gave me this output: `Could not find class 'android.widget.ThemedSpinnerAdapter', referenced from method android.support.v7.widget.AppCompatSpinner$DropDownAdapter.<init>`
See also https://stackoverflow.com/questions/34014843/could-not-find-class-android-widget-themedspinneradapter-android-studio/37050929 which does not have a proper solution...

Reply:
Does the same problem exist for the maxspeed quest? (I.e. go to Myanmar and download the quest there, the spinner should show)

Reply:
As I said: For the max speed quest, everything worked fine (I just tested it again...)